std/experimental/allocator/package.d

```diff
@@ -1484,10 +1494,18 @@ exception if the copy operation throws.
 T[] makeArray(T, Allocator)(auto ref Allocator alloc, size_t length)
 {
     if (!length) return null;
-    auto m = alloc.allocate(T.sizeof * length);
-    if (!m.ptr) return null;
-    alias U = Unqual!T;
-    return () @trusted { return cast(T[]) uninitializedFillDefault(cast(U[]) m); }();
+    static if (isInitAllZeroBits!T && is(typeof(alloc.allocateZeroed(length))))
```

Comment:
This also needs to check that the element is actually constructable (i.e. doesn't have `@disable this;`)

Reply:
That wouldn't change the result because the other branch (i.e. the current implementation) uses `uninitializedFillDefault` which ensures the array is filled with bitwise copies of `T.init` and doesn't care about `@disable this()`.